plugins/inspect/os_inspector.rb

```diff
@@ -31,53 +31,58 @@ def strip_arch_from_name(name)
   end
 
   # checks for additional version information like Beta or RC
-  def get_additional_version(system)
-    issue = system.read_file("/etc/issue")
+  def get_additional_version
+    issue = @system.read_file("/etc/issue")
     special_version = issue.scan(/Beta\d+|RC\d|GMC\d*/).first if issue
 
     special_version ? " #{special_version.gsub(/[0-9]{1,2}/," \\0")}" : ""
   end
 
-  def inspect(system, description, _filter, _options = {})
-    system.check_requirement("cat", "--version") if system.is_a?(RemoteSystem)
+  def initialize(system, description)
+    @system = system
+    @description = description
+  end
+
+  def inspect(_filter, _options = {})
+    @system.check_requirement("cat", "--version") if @system.is_a?(RemoteSystem)
 
-    os = get_os(system)
+    os = get_os
     if os
-      os.architecture = get_arch(system)
-      os.version += get_additional_version(system)
+      os.architecture = get_arch
+      os.version += get_additional_version
     else
       raise Machinery::Errors::UnknownOs
     end
 
-    description.os = os
+    @description.os = os
   end
 
-  def summary(description)
-    "Found operating system '#{description.os.name}' version '#{description.os.version}'."
+  def summary
+    "Found operating system '#{@description.os.name}' version '#{@description.os.version}'."
   end
 
   private
 
-  def get_os(system)
+  def get_os
     # Use os-release file by default
-    os = get_os_from_os_release(system)
+    os = get_os_from_os_release
 
     # Fall back to SuSE-release file
     if !os
-      os = get_os_from_suse_release(system)
+      os = get_os_from_suse_release
     end
 
     # Fall back to redhat-release file
     if !os
-      os = get_os_from_redhat_release(system)
+      os = get_os_from_redhat_release
     end
 
     os
   end
 
   # check for freedesktop standard: /etc/os-release
-  def get_os_from_os_release(system)
-    os_release = system.read_file("/etc/os-release")
+  def get_os_from_os_release
```

Comment:
Assignment Branch Condition size for get_os_from_os_release is too high. [21.47/15]